samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -900,26 +882,46 @@ static int configure_streams(void)
 		}
 
 		printk("Configured source stream[%zu]\n", i);
-		configured_stream_count++;
+		configured_source_stream_count++;
 	}
 
 	return 0;
 }
 
 static int create_group(void)
 {
-	struct bt_audio_unicast_group_stream_param stream_params[ARRAY_SIZE(streams)];
+	const size_t params_count = MAX(configured_sink_stream_count,
+					configured_source_stream_count);
+	struct bt_audio_unicast_group_stream_pair_param pair_params[params_count];
+	struct bt_audio_unicast_group_stream_param stream_params[configured_stream_count];
 	struct bt_audio_unicast_group_param param;
 	int err;
 
 	for (size_t i = 0U; i < configured_stream_count; i++) {
 		stream_params[i].stream = &streams[i];
 		stream_params[i].qos = &codec_configuration.qos;
-		stream_params[i].dir = stream_dir(stream_params[i].stream);
 	}
 
-	param.params = stream_params;
-	param.params_count = configured_stream_count;
+	for (size_t i = 0U; i < params_count; i++) {
+		struct bt_audio_unicast_group_stream_param *stream_param;
+
+		if (i < configured_sink_stream_count) {
+			stream_param = &stream_params[i];
+			pair_params[i].tx_param = stream_param;
```

Comment:
```suggestion
			pair_params[i].tx_param = &stream_params[i];
```
and subsequently removve the `struct bt_audio_unicast_group_stream_param *stream_param;`